File: src/main/java/rbq/lycoris/client/font/FontUtil.java
package rbq.lycoris.client.font;

import rbq.lycoris.client.utils.MinecraftInstance;
import rbq.lycoris.client.wrapper.wrappers.util.ResourceLocation;

import java.awt.*;


public class FontUtil extends MinecraftInstance {

    public static Font getFontFromTTF(ResourceLocation fontLocation, float fontSize, int fontType) {
        Font output = null;
        try {
            output = Font.createFont(fontType, mc.getResourceManager().getResource(fontLocation).getInputStream());
            output = output.deriveFont(fontSize);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return output;
    }
}

File: src/main/kotlin/rbq/lycoris/client/font/FontUtil.kt
package rbq.lycoris.client.font

import rbq.lycoris.client.utils.MinecraftInstance
import rbq.lycoris.client.wrapper.wrappers.util.ResourceLocation
import java.awt.Font
import java.awt.FontFormatException
import java.io.IOException

object FontUtil : MinecraftInstance() {
    @JvmStatic
    fun getFontFromTTF(fontLocation: ResourceLocation?, fontSize: Float, fontType: Int): Font {
        return try {
            Font.createFont(fontType, mc.resourceManager.getResource(fontLocation!!).inputStream).deriveFont(fontSize)
        } catch (e: FontFormatException) {
            throw RuntimeException(e)
        } catch (e: IOException) {
            throw RuntimeException(e)
        }
    }
}